src/app/components/Curation/HierarchicalGrid/index.tsx

```diff
@@ -140,9 +140,13 @@ const HiearchicalGrid = ({
                 <Promo.Body className="promo-paragraph" css={styles.body}>
                   {promo.description}
                 </Promo.Body>
-                <Promo.Timestamp className="promo-timestamp">
-                  {promo.lastPublished}
-                </Promo.Timestamp>
+                <React.Fragment>
+                  {!isLive ? (
+                    <Promo.Timestamp className="promo-timestamp">
+                      {promo.lastPublished}
+                    </Promo.Timestamp>
+                  ) : null}
+                </React.Fragment>
```

Comment:
Does this work?

```suggestion
                  {!isLive &&  (
                    <Promo.Timestamp className="promo-timestamp">
                      {promo.lastPublished}
                    </Promo.Timestamp>
                  )}
```

Reply:
Just commenting for the benefit of others who may read these comments - We did try the above (and below) changes initially but it unfortunately didn't work. The approach is used within the same file for other components, which we think may work due to those components being migrated/updated and typescriptified. 🤔

Reply:
😮 good to know

Reply:
Is it worth standardising the approach and using the ternary with `null` in these cases, e.g:
```
 {isLive ? (
   <Promo.Timestamp className="promo-timestamp">
     {promo.lastPublished}
   </Promo.Timestamp>
 ) : null}
```

Means you can get rid of the instances of `<React.Fragment>` and `<>`. I wonder is TypeScript just more cautious about returning `0` from these boolean conditionals: https://www.codemzy.com/blog/react-render-0-conditional, so it forces you to always return a valid React element, which null is technically.

Reply:
@amoore108 this is the approach we are deciding to go with, however this approach still doesn't allow us to render without the fragment wrapping it 🥲

Reply:
Is there an error thrown when you use the ternary without wrapping it in Fragment? I tried it locally and it seems to build ok.

Reply:
![Screenshot 2023-11-14 at 11 13 57](https://github.com/bbc/simorgh/assets/120198886/e7ecd2b3-5eda-4e34-88e7-4d423dda52a8)
This is with reinstalling node_modules and confirming that I have node 18 running locally 😭

Reply:
In case the screenshot is too small 

`2023-11-14 11:15:49.928 error {"file":"server/index.jsx","event":"ssr_request_failed","label":"EXPRESS","status":500,"message":"Cannot read properties of null (reading 'type')","url":"/kyrgyz","headers":{"host":"localhost:7080","user-agent":"Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:109.0) Gecko/20100101 Firefox/119.0","accept":"text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,*/*;q=0.8","accept-language":"en-GB,en;q=0.5","accept-encoding":"gzip, deflate, br","connection":"keep-alive","cookie":"ckns_privacy=july2019; ckns_policy=000; ckns_explicit=1; ckns_mvt=5b5ef44a-4899-4aff-8b87-8c2df9c8f78b; atuserid={%22val%22:%2256d1790e-3ee2-4e69-9f8f-7a4418e673c3%22}; cwr_u=088fb2d0-5769-4394-9d87-bd85f84333f8; comScore=amp-fA8_o5uaNxMUFly_YNvJ_Q; ecos.dt=1692964294930; cwr_s=eyJzZXNzaW9uSWQiOiI2M2Q1YzE4ZC1jYjE0LTRmNmQtOTI2NS1iZDM5YzY2ZWExOTAiLCJyZWNvcmQiOnRydWUsImV2ZW50Q291bnQiOjQ4LCJwYWdlIjp7InBhZ2VJZCI6Ii9reXJneXoiLCJwYXJlbnRQYWdlSWQiOiIvc2VyYmlhbi9jeXIvc3JiaWphLTY3MjczMzUzIiwiaW50ZXJhY3Rpb24iOjIsInN0YXJ0IjoxNjk5OTYwMjYxMDYxfX0=; _cb=BbF5HRCmsxfQv6vZv; _chartbeat2=.1672316771656.1699960261389.0111100001100011.Du3ovTCakF_7Bjtj_nDf9fSVDXEupS.2; _cb_svref=null","upgrade-insecure-requests":"1","sec-fetch-dest":"document","sec-fetch-mode":"navigate","sec-fetch-site":"none","sec-fetch-user":"?1"}}
2023-11-14 11:15:49.929 error {"file":"server/index.jsx","event":"ssr_request_failed","label":"EXPRESS","status":500,"message":"Cannot read properties of undefined (reading 'split')","url":"/kyrgyz","headers":{"host":"localhost:7080","user-agent":"Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:109.0) Gecko/20100101 Firefox/119.0","accept":"text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,*/*;q=0.8","accept-language":"en-GB,en;q=0.5","accept-encoding":"gzip, deflate, br","connection":"keep-alive","cookie":"ckns_privacy=july2019; ckns_policy=000; ckns_explicit=1; ckns_mvt=5b5ef44a-4899-4aff-8b87-8c2df9c8f78b; atuserid={%22val%22:%2256d1790e-3ee2-4e69-9f8f-7a4418e673c3%22}; cwr_u=088fb2d0-5769-4394-9d87-bd85f84333f8; comScore=amp-fA8_o5uaNxMUFly_YNvJ_Q; ecos.dt=1692964294930; cwr_s=eyJzZXNzaW9uSWQiOiI2M2Q1YzE4ZC1jYjE0LTRmNmQtOTI2NS1iZDM5YzY2ZWExOTAiLCJyZWNvcmQiOnRydWUsImV2ZW50Q291bnQiOjQ4LCJwYWdlIjp7InBhZ2VJZCI6Ii9reXJneXoiLCJwYXJlbnRQYWdlSWQiOiIvc2VyYmlhbi9jeXIvc3JiaWphLTY3MjczMzUzIiwiaW50ZXJhY3Rpb24iOjIsInN0YXJ0IjoxNjk5OTYwMjYxMDYxfX0=; _cb=BbF5HRCmsxfQv6vZv; _chartbeat2=.1672316771656.1699960261389.0111100001100011.Du3ovTCakF_7Bjtj_nDf9fSVDXEupS.2; _cb_svref=null","upgrade-insecure-requests":"1","sec-fetch-dest":"document","sec-fetch-mode":"navigate","sec-fetch-site":"none","sec-fetch-user":"?1"}}`

Reply:
Ah I can see why. Its the `Promo` component doing some manipulation with the `children` passed into it: https://github.com/bbc/simorgh/blob/d03d06e0a0e3025b46c1546d297a3f4da93ff0e9/src/app/legacy/components/Promo/index.jsx#L32-L35

So it doesn't handle a `null` child in the same way React normally would.

Reply:
Adding `children.filter(Boolean)` to line 34 there makes it work. That will filter out any falsy values.

Reply:
Thanks @amoore108 🙌